In [3]:
#Imorting Libraries

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns

In [55]:
#importing datasets

In [4]:
data =pd.read_csv("TRAIN.csv")
data1=pd.read_csv('TEST.csv')

In [5]:
data.head()

idx    UNIXTime                    Data      Time  Radiation  Temperature  \
0    0  1480107904  11/25/2016 12:00:00 AM  11:05:04     288.44           46   
1    1  1472818508    9/2/2016 12:00:00 AM  02:15:08       2.79           50   
2    2  1475804719   10/6/2016 12:00:00 AM  15:45:19     118.05           54   
3    3  1482533149  12/23/2016 12:00:00 AM  12:45:49     853.17           58   
4    4  1481883019  12/16/2016 12:00:00 AM  00:10:19       1.24           42   

   Pressure  Humidity  WindDirection(Degrees)  Speed TimeSunRise TimeSunSet  
0     30.48       101                  129.84  13.50    06:37:00   17:42:00  
1     30.42        75                  173.90   6.75    06:07:00   18:37:00  
2     30.42       100                    7.35   1.12    06:15:00   18:07:00  
3     30.44        57                   81.67  11.25    06:54:00   17:50:00  
4     30.24       103                  171.13   2.25    06:50:00   17:46:00

In [6]:
data.describe().transpose()

count          mean           std           min  \
idx                     26149.0  1.307400e+04  7.548710e+03  0.000000e+00   
UNIXTime                26149.0  1.478046e+09  3.004309e+06  1.472724e+09   
Radiation               26149.0  2.086210e+02  3.160897e+02  1.110000e+00   
Temperature             26149.0  5.113511e+01  6.212018e+00  3.500000e+01   
Pressure                26149.0  3.042301e+01  5.467877e-02  3.019000e+01   
Humidity                26149.0  7.497094e+01  2.600873e+01  8.000000e+00   
WindDirection(Degrees)  26149.0  1.433682e+02  8.342419e+01  9.000000e-02   
Speed                   26149.0  6.262052e+00  3.503488e+00  0.000000e+00   

                                 25%           50%           75%           max  
idx                     6.537000e+03  1.307400e+04  1.961100e+04  2.614800e+04  
UNIXTime                1.475544e+09  1.478029e+09  1.480473e+09  1.483264e+09  
Radiation               1.230000e+00  2.720000e+00  3.602600e+02  1.601260e+03  
Temperature             4.600000e+01  5.000000e+01  5.500000e+01  7.100000e+01  
Pressure                3.040000e+01  3.043000e+01  3.046000e+01  3.056000e+01  
Humidity                5.600000e+01  8.500000e+01  9.700000e+01  1.030000e+02  
WindDirection(Degrees)  8.158000e+01  1.473200e+02  1.792400e+02  3.599400e+02  
Speed                   3.370000e+00  5.620000e+00  7.870000e+00  4.050000e+01

In [7]:
data1.describe().transpose()

count          mean            std           min  \
idx                     6537.0  3.268000e+03    1887.213687  0.000000e+00   
UNIXTime                6537.0  1.481745e+09  636282.630100  1.480586e+09   
Temperature             6537.0  4.783632e+01       4.847630  3.400000e+01   
Pressure                6537.0  3.037649e+01       0.078843  3.019000e+01   
Humidity                6537.0  8.161328e+01      21.306379  2.000000e+01   
WindDirection(Degrees)  6537.0  1.788448e+02      87.664437  1.700000e-01   
Speed                   6537.0  6.738943e+00       4.788839  0.000000e+00   

                                 25%           50%           75%           max  
idx                     1.634000e+03  3.268000e+03  4.902000e+03  6.536000e+03  
UNIXTime                1.481301e+09  1.481792e+09  1.482285e+09  1.482775e+09  
Temperature             4.500000e+01  4.700000e+01  5.100000e+01  6.200000e+01  
Pressure                3.032000e+01  3.038000e+01  3.043000e+01  3.056000e+01  
Humidity                7.200000e+01  9.300000e+01  9.600000e+01  1.030000e+02  
WindDirection(Degrees)  1.261100e+02  1.700100e+02  2.365200e+02  3.599000e+02  
Speed                   3.370000e+00  5.620000e+00  9.000000e+00  4.050000e+01

In [56]:
#Converting into Data into Date and Time

In [8]:
import re 
data['Timenew']=re.sub('[^0-1]', '', str(data['Time']))

In [9]:
data.head()

idx    UNIXTime                    Data      Time  Radiation  Temperature  \
0    0  1480107904  11/25/2016 12:00:00 AM  11:05:04     288.44           46   
1    1  1472818508    9/2/2016 12:00:00 AM  02:15:08       2.79           50   
2    2  1475804719   10/6/2016 12:00:00 AM  15:45:19     118.05           54   
3    3  1482533149  12/23/2016 12:00:00 AM  12:45:49     853.17           58   
4    4  1481883019  12/16/2016 12:00:00 AM  00:10:19       1.24           42   

   Pressure  Humidity  WindDirection(Degrees)  Speed TimeSunRise TimeSunSet  \
0     30.48       101                  129.84  13.50    06:37:00   17:42:00   
1     30.42        75                  173.90   6.75    06:07:00   18:37:00   
2     30.42       100                    7.35   1.12    06:15:00   18:07:00   
3     30.44        57                   81.67  11.25    06:54:00   17:50:00   
4     30.24       103                  171.13   2.25    06:50:00   17:46:00   

                                  Timenew  
0  01100101011100101101011001010010001001  
1  01100101011100101101011001010010001001  
2  01100101011100101101011001010010001001  
3  01100101011100101101011001010010001001  
4  01100101011100101101011001010010001001

In [10]:
data1['Timenew']=re.sub('[^0-1]', '', str(data1['Time']))

In [11]:
data.head()

idx    UNIXTime                    Data      Time  Radiation  Temperature  \
0    0  1480107904  11/25/2016 12:00:00 AM  11:05:04     288.44           46   
1    1  1472818508    9/2/2016 12:00:00 AM  02:15:08       2.79           50   
2    2  1475804719   10/6/2016 12:00:00 AM  15:45:19     118.05           54   
3    3  1482533149  12/23/2016 12:00:00 AM  12:45:49     853.17           58   
4    4  1481883019  12/16/2016 12:00:00 AM  00:10:19       1.24           42   

   Pressure  Humidity  WindDirection(Degrees)  Speed TimeSunRise TimeSunSet  \
0     30.48       101                  129.84  13.50    06:37:00   17:42:00   
1     30.42        75                  173.90   6.75    06:07:00   18:37:00   
2     30.42       100                    7.35   1.12    06:15:00   18:07:00   
3     30.44        57                   81.67  11.25    06:54:00   17:50:00   
4     30.24       103                  171.13   2.25    06:50:00   17:46:00   

                                  Timenew  
0  01100101011100101101011001010010001001  
1  01100101011100101101011001010010001001  
2  01100101011100101101011001010010001001  
3  01100101011100101101011001010010001001  
4  01100101011100101101011001010010001001

In [12]:
x=data[['UNIXTime','Temperature','Pressure','Humidity','WindDirection(Degrees)','Speed']]
y=data[['Radiation']]

In [13]:
x_test=data1[['UNIXTime','Temperature','Pressure','Humidity','WindDirection(Degrees)','Speed']]

In [57]:
#Creating train and validation datasets

In [14]:
from sklearn.model_selection import train_test_split
xtrain,xvalid,ytrain,yvalid=train_test_split(x,y,test_size=0.3,random_state=1)

Polynomial features allow the use of simpler modeling algorithms as some of the complexity of interpreting the input variables and their relationships is pushed back to the data preparation stage.

In [15]:
from sklearn.preprocessing import PolynomialFeatures
pf=PolynomialFeatures(degree=2)
xtrain=pf.fit_transform(xtrain)
xvalid=pf.fit_transform(xvalid)
x_test=pf.fit_transform(x_test)

In [58]:
#Predicting with Linear Regression

In [16]:
from sklearn.linear_model import LinearRegression
lr=LinearRegression()

In [25]:
lr.fit(xtrain,ytrain)

LinearRegression()

In [26]:

ypred=lr.predict(xvalid)

In [27]:
from sklearn.metrics import mean_squared_error as mse
mse(yvalid,ypred)

32670.09289711875

In [28]:
np.sqrt(mse(yvalid,ypred))

180.74870095554974

After using Linear Regression we got Mean Squared error of 180.7487

In [29]:
#Predicting on the test dataset
yfinal=lr.predict(x_test)
yfinal=np.abs(yfinal)
yfinal

array([[ 65.9012198 ],
       [156.57664149],
       [ 87.35229703],
       ...,
       [  7.36276944],
       [  9.59942898],
       [  7.88116493]])

In [30]:
yfinal=pd.DataFrame(yfinal)
yfinal.to_csv("output1.csv")

In [59]:
#Predicting with Random Forest Regressor

In [31]:
from sklearn.ensemble import RandomForestRegressor
rfr=RandomForestRegressor(max_depth=110,max_features=9)
rfr.fit(xtrain,ytrain)

C:\Users\Shree\Anaconda331\lib\site-packages\ipykernel_launcher.py:3: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  This is separate from the ipykernel package so we can avoid doing imports until


RandomForestRegressor(max_depth=110, max_features=9)

In [32]:
y_pred=rfr.predict(xvalid)

In [33]:
mse(yvalid,y_pred)

13811.053672025599

In [34]:
np.sqrt(mse(yvalid,y_pred))

117.52043937981851

After using Random Forest Regressor we got Mean Squared error of 117.5204

In [35]:
y_final=rfr.predict(x_test)
print(y_final)
y_final=pd.DataFrame(y_final)

[ 63.0399 107.1892  64.859  ...  19.3401  10.2517  13.0742]


In [36]:
y_final.to_csv("output.csv")

In [37]:
y_final.head()

0
0   63.0399
1  107.1892
2   64.8590
3   92.1788
4   66.8572

In [43]:
#!pip install catboost

In [41]:
from catboost import CatBoostRegressor
cb =CatBoostRegressor(depth=12,iterations=500,learning_rate=0.02)

In [42]:
cb.fit(xtrain,ytrain)

0:	learn: 313.1679000	total: 764ms	remaining: 6m 21s
1:	learn: 309.1302600	total: 1.26s	remaining: 5m 14s
2:	learn: 305.0705141	total: 1.72s	remaining: 4m 44s
3:	learn: 301.3160997	total: 2.15s	remaining: 4m 27s
4:	learn: 297.5410345	total: 2.59s	remaining: 4m 16s
5:	learn: 293.9238916	total: 3.04s	remaining: 4m 10s
6:	learn: 290.2657418	total: 3.47s	remaining: 4m 4s
7:	learn: 286.6580900	total: 3.88s	remaining: 3m 58s
8:	learn: 283.1446334	total: 4.34s	remaining: 3m 57s
9:	learn: 279.8219002	total: 4.77s	remaining: 3m 53s
10:	learn: 276.4412571	total: 5.23s	remaining: 3m 52s
11:	learn: 273.1874813	total: 5.64s	remaining: 3m 49s
12:	learn: 269.9504998	total: 6.08s	remaining: 3m 47s
13:	learn: 266.8682919	total: 6.53s	remaining: 3m 46s
14:	learn: 263.9303355	total: 6.97s	remaining: 3m 45s
15:	learn: 260.9325031	total: 7.43s	remaining: 3m 44s
16:	learn: 258.0138651	total: 7.82s	remaining: 3m 42s
17:	learn: 255.2261073	total: 8.29s	remaining: 3m 42s
18:	learn: 252.5141791	total: 8.83s	rem

153:	learn: 138.5783811	total: 1m 4s	remaining: 2m 23s
154:	learn: 138.3407220	total: 1m 4s	remaining: 2m 23s
155:	learn: 138.1403040	total: 1m 4s	remaining: 2m 23s
156:	learn: 137.9143724	total: 1m 5s	remaining: 2m 22s
157:	learn: 137.7167167	total: 1m 5s	remaining: 2m 22s
158:	learn: 137.5075827	total: 1m 6s	remaining: 2m 21s
159:	learn: 137.3027441	total: 1m 6s	remaining: 2m 21s
160:	learn: 137.1021761	total: 1m 6s	remaining: 2m 20s
161:	learn: 136.9221318	total: 1m 7s	remaining: 2m 20s
162:	learn: 136.7057014	total: 1m 7s	remaining: 2m 19s
163:	learn: 136.4788083	total: 1m 8s	remaining: 2m 19s
164:	learn: 136.2817616	total: 1m 8s	remaining: 2m 18s
165:	learn: 136.0760462	total: 1m 8s	remaining: 2m 18s
166:	learn: 135.8990994	total: 1m 9s	remaining: 2m 18s
167:	learn: 135.6962270	total: 1m 9s	remaining: 2m 17s
168:	learn: 135.4526771	total: 1m 10s	remaining: 2m 17s
169:	learn: 135.2628240	total: 1m 10s	remaining: 2m 16s
170:	learn: 135.1158333	total: 1m 10s	remaining: 2m 16s
171:	le

301:	learn: 119.5056437	total: 2m 4s	remaining: 1m 21s
302:	learn: 119.4220346	total: 2m 4s	remaining: 1m 21s
303:	learn: 119.3757898	total: 2m 5s	remaining: 1m 20s
304:	learn: 119.2863992	total: 2m 5s	remaining: 1m 20s
305:	learn: 119.2270718	total: 2m 6s	remaining: 1m 19s
306:	learn: 119.1130799	total: 2m 6s	remaining: 1m 19s
307:	learn: 119.0183859	total: 2m 6s	remaining: 1m 19s
308:	learn: 118.9263097	total: 2m 7s	remaining: 1m 18s
309:	learn: 118.8512786	total: 2m 7s	remaining: 1m 18s
310:	learn: 118.7978086	total: 2m 8s	remaining: 1m 17s
311:	learn: 118.7330652	total: 2m 8s	remaining: 1m 17s
312:	learn: 118.6673430	total: 2m 9s	remaining: 1m 17s
313:	learn: 118.5775260	total: 2m 9s	remaining: 1m 16s
314:	learn: 118.5072545	total: 2m 9s	remaining: 1m 16s
315:	learn: 118.4259237	total: 2m 10s	remaining: 1m 15s
316:	learn: 118.3353798	total: 2m 10s	remaining: 1m 15s
317:	learn: 118.2599275	total: 2m 11s	remaining: 1m 15s
318:	learn: 118.1972442	total: 2m 11s	remaining: 1m 14s
319:	l

451:	learn: 108.7196913	total: 3m 3s	remaining: 19.5s
452:	learn: 108.6624920	total: 3m 4s	remaining: 19.1s
453:	learn: 108.6152531	total: 3m 4s	remaining: 18.7s
454:	learn: 108.5315921	total: 3m 5s	remaining: 18.3s
455:	learn: 108.4606158	total: 3m 5s	remaining: 17.9s
456:	learn: 108.4135467	total: 3m 5s	remaining: 17.5s
457:	learn: 108.3631577	total: 3m 6s	remaining: 17.1s
458:	learn: 108.3062070	total: 3m 6s	remaining: 16.7s
459:	learn: 108.2790435	total: 3m 7s	remaining: 16.3s
460:	learn: 108.2214987	total: 3m 7s	remaining: 15.9s
461:	learn: 108.1607503	total: 3m 7s	remaining: 15.5s
462:	learn: 108.0936885	total: 3m 8s	remaining: 15s
463:	learn: 108.0279027	total: 3m 8s	remaining: 14.6s
464:	learn: 107.9681900	total: 3m 9s	remaining: 14.2s
465:	learn: 107.9029582	total: 3m 9s	remaining: 13.8s
466:	learn: 107.8218913	total: 3m 10s	remaining: 13.4s
467:	learn: 107.7736969	total: 3m 10s	remaining: 13s
468:	learn: 107.7081352	total: 3m 11s	remaining: 12.6s
469:	learn: 107.6622455	total

In [44]:
yyy=cb.predict(xvalid)

In [45]:
yyyy=cb.predict(x_test)

In [46]:
yyyy=pd.DataFrame(yyyy)
yyyy

0
0     117.609368
1     132.208395
2     115.394299
3     118.564981
4     113.710728
...          ...
6532   16.708999
6533   15.945556
6534   16.546601
6535   16.082982
6536   27.382877

[6537 rows x 1 columns]

In [47]:
xt,xf,yt,yf=train_test_split(xvalid,yvalid,test_size=0.0005,random_state=7)

In [48]:
from xgboost import XGBRFRegressor
xgbr =XGBRFRegressor(random_state=7,max_depth=330,colsample_bynode=0.62,colsample_bytree=0.85,colsample_bylevel=1,subsample=.8,
                     n_estimators=95,reg_alpha=1.2,min_child_weight=0.6)
xgbr.fit(xtrain,ytrain)

[15:22:25] WARNING: C:/Jenkins/workspace/xgboost-win64_release_0.90/src/objective/regression_obj.cu:152: reg:linear is now deprecated in favor of reg:squarederror.


XGBRFRegressor(colsample_bynode=0.62, colsample_bytree=0.85, max_depth=330,
               min_child_weight=0.6, n_estimators=95, random_state=7,
               reg_alpha=1.2)

In [49]:
final=xgbr.predict(xvalid)

In [50]:
mse(yvalid,final)

15109.585158645392

In [51]:
np.sqrt(mse(yvalid,final))

122.9210525444905

In [52]:
finalq=xgbr.predict(x_test)
finalq=pd.DataFrame(finalq)

In [53]:
finalq.to_csv("finalq.csv")

In [54]:
finalq.head(10)

0
0  43.571255
1  96.613213
2  59.155369
3  63.744553
4  52.118168
5  63.160225
6   5.633932
7  20.645485
8  40.464451
9  30.340767